### ISSUE TITLE
sample_fmts

### ISSUE THREAD

**Issue by lucemia** [livingbio/typed-ffmpeg#219]
sample_fmts

```
name   depth
u8        8 
s16      16 
s32      32 
flt      32 
dbl      64 
u8p       8 
s16p     16 
s32p     32 
fltp     32 
dblp     64 
s64      64 
s64p     64 
```

_Originally posted by @lucemia in https://github.com/livingbio/typed-ffmpeg/issues/214#issuecomment-1902001356_

**Comment by Omarabdalgwad** [comment]
Hi @lucemia, for sample_fmt , are you thinking of using Literal from typing to restrict the values , or would you prefer keeping it as str with a list of allowed format options ?

**Comment by lucemia** [comment]
@Omarabdalgwad 

I’m not totally sure yet. Using Literal is attractive for type safety and autocomplete, but being too restrictive might hurt usability, especially if FFmpeg introduces new sample formats or if different builds expose additional ones.

A hybrid approach could give us the best of both worlds: keep flexibility while still providing helpful typing hints for the common formats.
```
from typing import Literal, Union

SampleFmtLiteral = Literal[
    "u8", "s16", "s32", "flt", "dbl",
    "u8p", "s16p", "s32p", "fltp", "dblp",
    "s64", "s64p",
]

SampleFmt = Union[SampleFmtLiteral, str]
```
This way we get IDE support and basic validation guidance, while still allowing advanced or future FFmpeg formats without breaking the API.

**Comment by Omarabdalgwad** [comment]
@lucemia 
Makes sense to me , I like this hybrid approach ,keeping it flexible while still having typing hints (maybe as a TypeAlias) seems like a nice balance

```bash
from typing import Literal, TypeAlias

SampleFmtLiteral: TypeAlias = Literal[
    "u8", "s16", "s32", "flt", "dbl",
    "u8p", "s16p", "s32p", "fltp", "dblp",
    "s64", "s64p",]

SampleFmt: TypeAlias = SampleFmtLiteral | str
```
I can draft a PR if you’re happy with this

**Comment by lucemia** [comment]
@Omarabdalgwad PR is always welcome!
